### PR TITLE
LPS-92806 Layout by experience

### PR DESCRIPTION
**Pull request by andresfulla**

### CODE REVIEW

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/utils/FragmentsEditorUpdateUtils.es.js [hunk 1]
```diff
@@ -230,27 +230,32 @@ function updateIn(object, keyPath, updater, defaultValue) {
...
- * @param {!string} classPK
- * @param {!Object} data
- * @param {!Array} fragmentEntryLinkIds
+ * @param {!object} config
+ * @param {!string} config.updateLayoutPageTemplateDataURL
+ * @param {!string} config.portletNamespace
+ * @param {!string} config.classNameId
+ * @param {!string} config.classPK
+ * @param {!Object} config.data
+ * @param {!Array} config.fragmentEntryLinkIds
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/utils/FragmentsEditorUpdateUtils.es.js]: inform about removing purpose

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/segmentsExperiences.es.js [hunk 2]
```diff
@@ -9,13 +10,151 @@ const EDIT_SEGMENTS_EXPERIENCE_URL = '/segments.segmentsexperience/update-segmen
...
+ * @param {!object} state
+ * @param {!string} segmentsExperienceId
+ * @returns {Promise}
+ */
+function _switchLayoutDataList(state, segmentsExperienceId) {
+	let nextState = state;
+	return new Promise((resolve, reject) => {
+		try {
+
+		// this makes sure the server gets the latest changes of the layout belonging to the experience we are leaving
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/segmentsExperiences.es.js]: No comments here

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/segmentsExperiences.es.js [hunk 3]
```diff
@@ -9,13 +10,151 @@ const EDIT_SEGMENTS_EXPERIENCE_URL = '/segments.segmentsexperience/update-segmen
...
+								let noSingleReturnSegmentedLayout = segmentedLayout;
+								if (segmentedLayout.segmentsExperienceId === prevSegmentsExperienceId) {
+									noSingleReturnSegmentedLayout = Object.assign(
+										{},
+										segmentedLayout,
+										{
+											layoutData: prevLayout
+										}
+									);
+								}
```
[the remark this answers is not in the document]
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/segmentsExperiences.es.js]: variable name, not nice

**Pull request closed without merging**